common/src/main/java/io/druid/timeline/partition/NumberedPartitionChunk.java

```diff
@@ -58,19 +58,28 @@ public T getObject()
   @Override
   public boolean abuts(final PartitionChunk<T> other)
   {
-    return other instanceof NumberedPartitionChunk && other.getChunkNumber() == chunkNumber + 1;
+    if (other instanceof NumberedPartitionChunk) {
+      NumberedPartitionChunk<T> castedOther = (NumberedPartitionChunk<T>) other;
+      if (castedOther.getChunkNumber() < castedOther.chunks) {
+        return other.getChunkNumber() == chunkNumber + 1;
+      } else {
+        return other.getChunkNumber() > chunkNumber;
+      }
+    } else {
+      return false;
+    }
```

Comment:
should there be an additional check for "castedOther.chunks == this.chunks" or else false? I haven't put it because that was not the behavior before but it seems logical to put this check.
thoughts?